commands/diceRoller.js

```diff
@@ -124,13 +124,14 @@ function parseDiceCommand(command) {
   let dice = new Object();
   const commandCleaned = command.toLowerCase().split(' ').join('').split('r').join(''); // removes spaces and Rs
   let [amount, diceConfig] = commandCleaned.split('d');
-  let [sides, modifier] = diceConfig.split('+') || diceConfig.split('-');
+  let [sides, modifier] = diceConfig.includes('+') ? diceConfig.split('+') : diceConfig.split('-');
   dice.reroll = command.toLowerCase().includes('r'); //Boolean
   dice.add = diceConfig.includes('+'); //Boolean
   dice.subtract = diceConfig.includes('-'); //Boolean
   dice.amount = Number(amount); //Number
   dice.sides = Number(sides); //Number
-  dice.modifier = Number(modifier) || 0; // Sets the value of the modifier to 0 if not provided.
+  // I really don't like using the nullish operator inside of the Number method but this is the quickest fix for the time being, it seems weird to set 0 to a number when it already is one.
+  dice.modifier = Number(modifier ?? 0); // Sets the value of the modifier to 0 if not provided.
```

Comment:
Passing `0` to `Number` isn't as icky as you make it out to be, but if you want to avoid this (and actually be more robust overall), you could do it this way:

```suggestion
  dice.modifier = Number(modifiers);
  if (isNaN(dice.modifier)) {
  	dice.modifier = 0;
  }
```

Reply:
I'm going to hold off on this one for now because I have a slight idea of another way to take this, but I'm not sure if I want to go that route yet.